### PR TITLE
[numpy] fix cython

### PR DESCRIPTION
**Pull request by hzfan**
## Description ##
This allows numpy branch to run with cython.
Cython is enabled in CI as well.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- numpy-compatible cython ndarray and symbol
- enable cython in ci test

## Comments ##
- Thank @haojin2  @reminisce for reviewing and guidance.

### CODE REVIEW

File: python/mxnet/cython/ndarray.pyx [hunk 1]
```diff
@@ -64,21 +64,29 @@ cdef class NDArrayBase:
...
     _ndarray_cls = cls
 
 
-cdef NewArray(NDArrayHandle handle, int stype=-1):
+def _set_np_ndarray_class(cls):
+    global _np_ndarray_cls
+    _np_ndarray_cls = cls
+
+
+cdef NewArray(NDArrayHandle handle, int is_np_op, int stype=-1):
```
Comment (reminisce) [inline python/mxnet/cython/ndarray.pyx]: 1. Rename `is_np_op` to `is_np_array` for better readability.
2. Better not to change the original API. You can switch `is_np_array` with `stype` and default `is_np_array` to 0.

File: python/mxnet/cython/ndarray.pyx [hunk 2]
```diff
@@ -64,21 +64,29 @@ cdef class NDArrayBase:
...
     """Create a new array given handle"""
-    return _ndarray_cls(_ctypes.cast(<unsigned long long>handle, _ctypes.c_void_p), stype=stype)
+    if is_np_op:
```
Comment (reminisce) [inline python/mxnet/cython/ndarray.pyx]: These four lines can be simplified into two:
```python
create_array_fn = _np_ndarray_cls if is_np_array else _ndarray_cls
return create_array_fn(_ctypes.cast(<unsigned long long>handle, _ctypes.c_void_p), stype=stype)
```

File: python/mxnet/cython/symbol.pyx [hunk 3]
```diff
@@ -84,19 +84,29 @@ cdef SymbolSetAttr(SymbolHandle handle, dict kwargs):
...
     _symbol_cls = cls
 
-cdef NewSymbol(SymbolHandle handle):
+
+def _set_np_symbol_class(cls):
+    global _np_symbol_cls
+    _np_symbol_cls = cls
+
+
+cdef NewSymbol(SymbolHandle handle, int is_np_op):
```
Comment (reminisce) [inline python/mxnet/cython/symbol.pyx]: `is_np_op` --> `is_np_sym`

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Pull request merged**